webapp/src/components/expanded_view/recording_info_prompt.tsx

```diff
@@ -0,0 +1,153 @@
+import React, {CSSProperties, useState, useEffect} from 'react';
+import moment from 'moment-timezone';
+
+import {RecordCircleOutlineIcon} from '@mattermost/compass-icons/components';
+
+import CompassIcon from 'src/components/icons/compassIcon';
+
+import {
+    CallRecordingState,
+} from 'src/types/types';
+
+import {
+    CallRecordingDisclaimerStrings,
+} from 'src/constants';
+
+import {
+    capitalize,
+} from 'src/utils';
+
+import InCallPrompt from './in_call_prompt';
+
+type Props = {
+    isHost: boolean,
+    recording?: CallRecordingState,
+    recordingMaxDuration: number,
+    onDecline: () => void;
+}
+
+const minutesLeftThreshold = 10;
+
+export default function RecordingInfoPrompt(props: Props) {
+    // This is needed to force a re-render to periodically check for recording
+    // duration.
+    const [, updateState] = useState({});
+    useEffect(() => {
+        const interval = setInterval(() => updateState({}), 30000);
+        return () => clearInterval(interval);
+    });
+
+    const getMinutesLeftBeforeEnd = () => {
+        if (!props.recording?.start_at) {
+            return 0;
+        }
+        const callDurationMinutes = (Date.now() - props.recording.start_at) / (1000 * 60);
+        return Math.round(props.recordingMaxDuration - callDurationMinutes);
+    };
+
+    const [dismissedAt, updateDismissedAt] = useState(0);
+
+    const [recordingWillEndSoon, updateRecordingWillEndSoon] = useState(0);
+    useEffect(() => {
+        if (!props.isHost || !props.recording || props.recording.start_at === 0 || props.recording.end_at !== 0 || recordingWillEndSoon !== 0) {
+            return;
+        }
+
+        if (getMinutesLeftBeforeEnd() <= minutesLeftThreshold) {
+            updateRecordingWillEndSoon(Date.now());
+        }
+    });
+
+    const hasRecEnded = props.recording?.end_at;
+
+    if (props.isHost && !hasRecEnded && recordingWillEndSoon > dismissedAt) {
+        return (
+            <InCallPrompt
+                icon={
+                    <CompassIcon
+                        icon='alert-outline'
+                        style={{
+                            fontSize: 18,
+                        }}
+                    />
+                }
+                iconFill='rgb(var(--dnd-indicator-rgb))'
+                iconColor='rgb(var(--dnd-indicator-rgb))'
+                header={`Your recording will end in ${getMinutesLeftBeforeEnd()} minutes`}
+                body={`Calls can be recorded for up to ${props.recordingMaxDuration} minutes.`}
```

Comment:
@justinegeffen Please help with the right content to show here.

Reply:
This reads well to me. I wonder if we should switch the body and header?

header={`Calls can be recorded for up to ${props.recordingMaxDuration} minutes.`}
body={`Your recording will end in ${getMinutesLeftBeforeEnd()} minutes`}

Reply:
Yep, we can do that. Let me update on the main PR. Thanks!